packages/salesforcedx-vscode-core/src/commands/functions/forceFunctionStart.ts

```diff
@@ -143,7 +147,10 @@ export class ForceFunctionStartExecutor extends SfdxCommandletExecutor<string> {
     );
 
     execution.stdoutSubject.subscribe(data => {
-      if (data.toString().includes('Debugger running on port')) {
+      const matches = String(data).match(runtimeDetection);
```

Comment:
Curious if this is the only option to find the function type? Any information about the language or the directory structure at the time of registering the function we could use instead?
I'm just trying to avoid the need to maintain a regex for any future changes in the output.

Reply:
That was my first approach, but I decide to change it. Since the build-pack is creating the environment seems like it is best suited to determine the runtime/debug settings.

Reply:
In that case, can we add an else condition for the line below where we check for matches and give a user warning that reads something like "Language could not be identified. 'Debug Invoke' may not work".

Reply:
I'll see if I can move that type of logic to the 'Debug Invoke' - it should not be visible/enabled if the language hasn't been determined, or at least produce the warning you mentioned.

Reply:
Also looks like there is enough business logic now to add a separate section of tests in forceFunctionStart.test.ts